ts/store/actions/deferred.ts

```diff
@@ -0,0 +1,49 @@
+/**
+ * Action types and action creators related to deferred actions.
+ * A deferred action is an action dispatched before the user login
+ * but actually has to be dispatched once the has been logged in.
+ * (e.g. navigation coming from push notification or from deep link happened
+ * while the app is not in foreground and/or the user is not logged in)
+ */
+
+import {
+  CLEAR_DEFERRED_ACTIONS,
+  DEFER_TO_LOGIN,
+  PUSH_TO_DEFERRED_ACTIONS
+} from "./constants";
+import { Action } from "./types";
+
+export interface DeferToLogin {
+  type: typeof DEFER_TO_LOGIN;
+  payload: Action;
+}
+
+export const deferToLogin = (action: Action): DeferToLogin => ({
```

Comment:
I think we need a better name for this action, before reading the explanation above I thought we were talking about deferring to the _next_ login - what about `deferActionAfterLogin`

Reply:
Ok, `deferActionAfterLogin` sounds better.